src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts

```diff
@@ -520,43 +526,37 @@ class RemoteMachineTrainingService implements TrainingService {
         return deferred.promise;
     }
 
-    private async launchTrialOnScheduledMachine(trialJobId: string, trialWorkingFolder: string, form: TrialJobApplicationForm,
+    private async launchTrialOnScheduledMachine(trialJobId: string, form: TrialJobApplicationForm,
         rmScheduleInfo: RemoteMachineScheduleInfo): Promise<void> {
         if (this.trialConfig === undefined) {
             throw new Error('trial config is not initialized');
         }
         const cudaVisibleDevice: string = rmScheduleInfo.cudaVisibleDevice;
-        const executor: ShellExecutor | undefined = this.trialExecutorMap.get(trialJobId);
-        if (executor === undefined) {
-            assert(false, 'ShellExecutor is undefined.');
-
-            // for lint
-            return;
-        }
+        const executor = await this.getExecutor(trialJobId);
         const trialJobDetail: RemoteMachineTrialJobDetail | undefined = this.trialJobsMap.get(trialJobId);
         if (trialJobDetail === undefined) {
             throw new Error(`Can not get trial job detail for job: ${trialJobId}`);
         }
 
         const trialLocalTempFolder: string = path.join(this.expRootDir, 'trials-local', trialJobId);
 
-        await executor.createFolder(trialWorkingFolder);
-        await executor.createFolder(unixPathJoin(trialWorkingFolder, '.nni'));
+        await executor.createFolder(trialJobDetail.workingDirectory);
+        await executor.createFolder(executor.joinPath(trialJobDetail.workingDirectory, '.nni'));
```

Comment:
Why not use a function support create folder recursively?

Reply:
fixed